lib/number.js

```diff
@@ -53,12 +53,29 @@ internals.Number = class extends Any {
 
     multiple(base) {
 
-        Hoek.assert(Hoek.isInteger(base), 'multiple must be an integer');
-        Hoek.assert(base > 0, 'multiple must be greater than 0');
+        const isRef = Ref.isRef(base);
 
-        return this._test('multiple', base, function (value, state, options) {
+        return this._test('multiple', base, (value, state, options) => {
 
-            if (value % base === 0) {
+            let divisor;
+            if (isRef) {
+                divisor = base(state.parent, options);
+                if (typeof divisor !== 'number') {
+                    return this.createError('number.ref', { ref: base.key }, state, options);
+                }
+
+                if (divisor === 0) {
+                    return this.createError('number.refZero', { ref: base.key }, state, options);
+                }
+            }
+            else {
+                divisor = base;
+                if ((typeof divisor !== 'number') || (divisor === 0)) {
```

Comment:
Are you asking for removing 

`if (divisor === 0) {`

If this is removed then It starts throwing error if the argument is zero.  The idea is to return if the number.multiple() test passes or not instead of throwing error. The idea is to validate gracefully instead of throwing errors.